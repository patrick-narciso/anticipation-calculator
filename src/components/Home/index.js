// eslint-disable-next-line no-unused-vars
import { dom } from 'isomorphic-jsx';
import { css } from 'emotion';
import { TextField } from '@Uikit';
import tokens from '@Uikit/Tokens';

const Home = () => {
  return (
    <Container>
      <Box>
        <Simulator>
          <SimulatorTitle>Simule sua Antecipação</SimulatorTitle>
          <TextField id="amount" type="text" label="Informe o valor da venda*" />
          <TextField
            id="installments"
            extraMessage="Máximo de 12 parcelas"
            type="number"
            label="Em quantas parcelas*"
          />
          <TextField id="mdr" type="text" label="Informe o percentual de MDR*" />
        </Simulator>
        <ResultSection>
          <ResultTitle>Você Receberá</ResultTitle>
          <ResultDays>
            <li>
              Amanhã: <strong>R$ 0,00</strong>
            </li>
            <li>
              Em 15 dias: <strong>R$ 0,00</strong>
            </li>
            <li>
              Em 30 dias: <strong>R$ 0,00</strong>
            </li>
            <li>
              Em 90 dias: <strong>R$ 0,00</strong>
            </li>
          </ResultDays>
        </ResultSection>
      </Box>
    </Container>
  );
};

const Container = ({ children }) => (
  <div
    class={css`
      display: flex;
      height: 100vh;
      align-items: center;
      justify-content: center;
    `}>
    {children}
  </div>
);

const Box = ({ children }) => (
  <div
    class={css`
      display: flex;
      border: 1px solid #d1dce3;
      box-sizing: border-box;
      border-radius: 4px;
      max-width: 608px;
      height: 400px;
      background: #ffffff;
    `}>
    {children}
  </div>
);

const Simulator = ({ children }) => (
  <section
    class={css`
      width: 377px;
      margin-left: 56px;
      margin-right: 70px;
    `}>
    {children}
  </section>
);

const SimulatorTitle = ({ children }) => (
  <h1
    class={css`
      margin-bottom: 22px;
      margin-top: 41px;
      font-weight: bold;
      font-size: 24px;
      line-height: 30px;
      color: ${tokens.brand('black')};
    `}>
    {children}
  </h1>
);

const ResultSection = ({ children }) => (
  <section
    class={css`
      width: 50%;
      background: rgba(209, 220, 227, 0.18);
      padding-left: 35px;
      padding-top: 70px;
    `}>
    {children}
  </section>
);

const ResultDays = ({ children }) => (
  <ul
    class={css`
      list-style: none;
      margin: 0;
      padding: 0;
      font-style: italic;
      line-height: 46px;
      font-size: ${tokens.fontSize('s')};
      color: ${tokens.color('blue', 'light')};
    `}>
    {children}
  </ul>
);

const ResultTitle = ({ children }) => (
  <p
    class={css`
      border-bottom: 1px solid rgba(93, 156, 236, 0.3);
      text-transform: uppercase;
      font-size: ${tokens.fontSize('s')};
      line-height: 30px;
      font-style: italic;
      font-weight: bold;
      width: 155px;
      color: ${tokens.color('blue', 'dark')};
    `}>
    {children}
  </p>
);

export default Home;
