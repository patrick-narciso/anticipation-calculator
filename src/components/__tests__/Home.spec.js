// eslint-disable-next-line no-unused-vars
import { dom } from 'isomorphic-jsx';
import Home from '@/components/Home';

test('Should be render Home Component correctly', () => {
  expect(<Home />).toMatchSnapshot();
});
