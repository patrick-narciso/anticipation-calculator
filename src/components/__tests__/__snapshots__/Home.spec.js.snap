// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be render Home Component correctly 1`] = `"<div class=\\"css-1ypwhrp-Container\\"><div class=\\"css-1wbqkmv-Box\\"><section class=\\"css-4dudyx-Simulator\\"><h1 class=\\"css-ejkvtm-SimulatorTitle\\">Simule sua Antecipação</h1><form id=\\"form\\"><div class=\\"css-11k5a09-wrapper\\"><label class=\\"css-18tgiz1-labelInput\\" for=\\"amount\\">Informe o valor da venda*</label><div class=\\"css-1rdjunr-inputWrapper\\"><div class=\\"css-memc06-wrapper\\"><input id=\\"amount\\" class=\\"css-1u7nkg2-fieldInput\\" name=\\"amount\\" type=\\"text\\" onchange=\\"undefined\\" value=\\"\\"></input></div></div></div><div class=\\"css-11k5a09-wrapper\\"><label class=\\"css-18tgiz1-labelInput\\" for=\\"installments\\">Em quantas parcelas*</label><div class=\\"css-1rdjunr-inputWrapper\\"><div class=\\"css-memc06-wrapper\\"><input id=\\"installments\\" class=\\"css-1u7nkg2-fieldInput\\" name=\\"installments\\" type=\\"number\\" onchange=\\"undefined\\" value=\\"\\"></input></div><span class=\\"css-1vd4a9j-extraInfo\\">Máximo de 12 parcelas</span></div></div><div class=\\"css-11k5a09-wrapper\\"><label class=\\"css-18tgiz1-labelInput\\" for=\\"mdr\\">Informe o percentual de MDR*</label><div class=\\"css-1rdjunr-inputWrapper\\"><div class=\\"css-memc06-wrapper\\"><input id=\\"mdr\\" class=\\"css-1u7nkg2-fieldInput\\" name=\\"mdr\\" type=\\"number\\" onchange=\\"undefined\\" value=\\"\\"></input></div></div></div></form></section><section class=\\"css-1lh4gfx-ResultSection\\"><p class=\\"css-tspkll-ResultTitle\\">Você Receberá</p><ul class=\\"css-srng7-ResultDays\\"><li id=\\"sei\\">Amanhã: <strong data-one-value=\\"0\\">R$ 0,00</strong></li><li>Em 15 dias: <strong data-fifteen-value=\\"0\\">R$ 0,00</strong></li><li>Em 30 dias: <strong data-thirty-value=\\"0\\">R$ 0,00</strong></li><li>Em 90 dias: <strong data-ninety-value=\\"0\\">R$ 0,00</strong></li></ul></section></div></div>"`;
